### PR TITLE
feat: update anthos-vmruntime example for ABM 1.12

### LINKED ISSUE #420: anthos-vmruntime example needs updating for VM Runtime 1.12 GA

### PR DESCRIPTION
**Pull request by apeabody**
#### Description
- Updates anthos-vmruntime for ABM 1.12

#### Change summary
anthos-vmruntime
- Cli `--cpu` is now `--vcpus`
- Updated expected output to 1.12 format
- Added `VirtualMachineDisk` cli example
- Updated .yaml files to VM Runtime v1 syntax
- Updated `vm.cluster.gke.io/vm-name` is now `kubevirt/vm`

### CODE REVIEW

**Review by Shabirmean — APPROVED** [review]
Thank you very much Andrew for updating this. LGTM!
I will try it ot once and merge it!

File: anthos-vmruntime/README.md [hunk 1]
```diff
@@ -100,10 +100,10 @@ using `kubectl apply -f vmruntime.yaml`.
 
 Validate that the `VMRuntime` is enabled.
 ```sh
-kubectl describe vmruntime vmruntime | grep Ready
+kubectl wait --for=jsonpath='{.status.ready}'=true --timeout=300s vmruntime vmruntime
 
 # expected output
-  Ready:                       true
+vmruntime.vm.cluster.gke.io/vmruntime condition met
 ```
```
Comment (Shabirmean) [inline anthos-vmruntime/README.md:107]: Can we have the `expected output` in a separate code block? So clicking the copy button on the other one doesn't copy this.

File: anthos-vmruntime/README.md [hunk 2]
```diff
@@ -159,33 +159,42 @@ create the VM using `kubectl apply -f pos-vm.yaml`.
...
-`VirtualMachine`, automatically triggers the creation of a
-`VirtualMachineInstance` resource which represents a _RUNNING_ instance of a VM.
+Creation of the VM requires two resources to be created: `VirtualMachineDisk` and
+`VirtualMachine`. `VirtualMachineDisk` is the _persistent disk_ where the
+contents of the **VM image** is imported into and is made bootable from.
+`VirtualMachine` is the **VM template** created based off of the image. The
+`VirtualMachineDisk` is mounted into the `VirtualMachine` before the VM is
+booted up. Creation of a `VirtualMachine`, automatically triggers the creation
+of a `VirtualMachineInstance` resource which represents a _RUNNING_ instance of
+a VM.
```
Comment (Shabirmean) [inline anthos-vmruntime/README.md:187]: I think we should add a few more details so readers are not confused of the use of `DataVolume` below. Also I have formatted it a bit.

```suggestion
Creation of the VM requires two resources to be created: `VirtualMachineDisk` and
`VirtualMachine`. 

- `VirtualMachineDisk` is the _persistent disk_ where the contents of the
**VM image** is imported into and is made bootable from. The `VirtualMachineDisk`
wraps around the `DataVolume` resource which can be used to monitor progress
of the import.

- `VirtualMachine` is the **VM template** created based off of the image. 
Creation of a `VirtualMachine`, automatically triggers the creation of a
`VirtualMachineInstance` resource which represents a _RUNNING_ instance of
a VM. The `VirtualMachineDisk` is mounted into the `VirtualMachine` before the VM is
booted up. 
```

File: anthos-vmruntime/pos-vm.yaml [hunk 3]
```diff
@@ -13,59 +13,36 @@
...
+      vcpus: 2
+    memory:
+      capacity: 4Gi
+  disks:
+  - boot: true
+    driver: virtio
+    virtualMachineDiskName: pos-vm-boot-dv
   interfaces:
   - name: eth0
+    default: true
```
Comment (Shabirmean) [inline anthos-vmruntime/pos-vm.yaml:46]: There seems to be 2 minor differences to what was generated in my cluster:
https://paste.googleplex.com/5887058592661504

1. The `VirtualMachineDisk` has a `storageClassName: local-shared` attribute
2. The API for `VirtualMachine ` is listed as `vm.cluster.gke.io/vm.cluster.gke.io/v1` 

Not sure if the second is a bug. Can we use the paste above to ensure it is consistent with what is output?

**Review by Shabirmean — CHANGES_REQUESTED** [review]
@apeabody - I did a second pass whilst trying out the steps again. I've made some suggestions and comments. Please take a look!

File: anthos-vmruntime/README.md (hunk 2 above)
Comment (apeabody) [reply anthos-vmruntime/README.md]: LGTM

File: anthos-vmruntime/README.md (hunk 1 above)
Comment (apeabody) [reply anthos-vmruntime/README.md:107]: For sure

File: anthos-vmruntime/pos-vm.yaml (hunk 3 above)
Comment (apeabody) [reply anthos-vmruntime/pos-vm.yaml:46]: I'll add #1, and as discussed let's leave @2 as-is.

File: anthos-vmruntime/README.md [hunk 4]
```diff
@@ -159,33 +161,47 @@ create the VM using `kubectl apply -f pos-vm.yaml`.
 kubectl virt create vm pos-vm \
 --boot-disk-size=80Gi \
 --memory=4Gi \
---cpu=2 \
+--vcpus=2 \
```
Comment (Shabirmean) [inline anthos-vmruntime/README.md:164]: Can you fix the formatting error in line 156?
`[/anthos-vmruntime/pos-vm.yaml](/pos-vm.yaml)` must be `[/anthos-vmruntime/pos-vm.yaml](pos-vm.yaml)`

**Review by Shabirmean — APPROVED** [review]
Thanks for addressing the comments. One minor request to fix a small bug in the README 🙏🏼

File: anthos-vmruntime/README.md (hunk 4 above)
Comment (apeabody) [reply anthos-vmruntime/README.md:164]: Done!

**Comment by Shabirmean** [comment]
Thank you very much @apeabody

**Pull request merged**